apps/commercetools/src/Editor/Field.tsx

```diff
@@ -0,0 +1,175 @@
+import * as React from 'react';
+import { Button } from '@contentful/forma-36-react-components';
+import tokens from '@contentful/forma-36-tokens';
+import get from 'lodash/get';
+import { css } from 'emotion';
+import { FieldExtensionSDK } from 'contentful-ui-extensions-sdk';
+import { ProductPreviews } from './ProductPreviews/ProductPreviews';
+import { CategoryPreviews } from './CategoryPreviews/CategoryPreviews';
+import { fetchProductPreviews } from '../api/fetchProductPreviews';
+import { fetchCategoryPreviews } from '../api/fetchCategoryPreviews';
+import logo from '../logo.svg';
+
+interface Props {
+  sdk: FieldExtensionSDK;
+}
+
+interface State {
+  value: string[];
+  editingDisabled: boolean;
+}
+
+const styles = {
+  sortable: css({
+    marginBottom: tokens.spacingM
+  }),
+  container: css({
+    display: 'flex'
+  }),
+  logo: css({
+    display: 'block',
+    width: '30px',
+    height: '30px',
+    marginRight: tokens.spacingM
+  })
+};
+
+function fieldValueToState(value?: string | string[]): string[] {
+  if (!value) {
+    return [];
+  }
+  return Array.isArray(value) ? value : [value];
+}
+
+function makeCTAText(fieldType: string, pickerMode: 'category' | 'product') {
+  const isArray = fieldType === 'Array';
+  const beingSelected =
+    pickerMode === 'category'
+      ? isArray
+        ? 'categories'
+        : 'category'
+      : isArray
+      ? 'products'
+      : 'product';
```

Comment:
Nit: if you put `a product` | `a category` you can simplify the last line

Reply:
Also, I'm pretty sure no linter will agree with the nested conditional expressions

Reply:
Done, linter's not complaining :sweat_smile: